src/types/graphs/viewgraph.ts

```diff
@@ -0,0 +1,286 @@
+import { Device } from "./../device"; // Import the Device class
+import { Edge } from "./../edge";
+import { DataGraph } from "./datagraph";
+import { Viewport } from "../..";
+
+export class ViewGraph {
+  private devices = new Map<number, Device>();
+  private edges = new Map<number, Edge>();
+  private idCounter = 1;
+  private datagraph: DataGraph;
+  viewport: Viewport;
+
+  constructor(datagraph: DataGraph, viewport: Viewport) {
+    this.datagraph = datagraph;
+    this.viewport = viewport;
+    datagraph.constructView(this);
```

Comment:
Since we'll have one view per layer, I think the viewgraph should construct itself